src/idl_gen_rust.cpp

```diff
@@ -698,6 +699,10 @@ class RustGenerator : public BaseGenerator {
   // an enum match function,
   // and an enum array of values
   void GenEnum(const EnumDef &enum_def) {
+    if (parser_.opts.no_leak_private_annotations) {
+      const auto is_private_access = enum_def.attributes.Lookup("private");
+      code_.SetValue("ACCESS_TYPE", is_private_access ? "pub(crate)": "pub");
```

Comment:
Instead of making ACCESS_VALUE pub and then overwriting, I think you should say

```cpp
const bool private = parser.opts.no_leak_private_annotations &&
    (enum_def.attributes.Lookup("private") != nullptr);
```
and call `SetValue("ACCESS_TYPE", ...)` unconditionally